### PR TITLE
add qaqc.py in core

### PR DESCRIPTION
**Pull request by ssuttles-usgs**
Separated out qaqc code in exo.py and eco.py into new qaqc.py in stglib/stglib/core. Affected files were updated, and standard tests (test_scripts.py) were ran to confirm functionality. Code formatting was checked with Black code formatter.

### CODE REVIEW

File: stglib/aqd/cdf2nc.py [hunk 1]
```diff
@@ -80,10 +80,9 @@ def cdf_to_nc(cdf_filename, atmpres=False):
 
     # should function this
     for var in VEL.data_vars:
-        VEL = exo.trim_max_diff(VEL, var)
-        VEL = exo.trim_min_diff(VEL, var)
-        VEL = exo.trim_min(VEL, var)
-        VEL = exo.trim_max(VEL, var)
+        VEL = qaqc.trim_min_diff(VEL, var)
```
Comment (dnowacki-usgs) [inline stglib/aqd/cdf2nc.py:83]: What happened to trim_max_diff?

File: stglib/exo.py [hunk 2]
```diff
@@ -577,210 +578,26 @@ def exo_qaqc(ds):
...
-        ds = trim_med_diff_pct(ds, var)
+        ds = qaqc.trim_med_diff_pct(ds, var)
 
-        ds = trim_bad_ens(ds, var)
+        ds = qaqc.trim_bad_ens(ds, var)
 
     for var in varlist:
-        ds = trim_by_any(ds, var)  # re-run and trim by other variables as necessary
+        ds = qaqc.trim_by_any(
+            ds, var
```
Comment (dnowacki-usgs) [inline stglib/exo.py:597]: This seems errant

File: stglib/exo.py [hunk 3]
```diff
@@ -577,210 +578,26 @@ def exo_qaqc(ds):
...
-                print(f"{var}: Trimming using valid {trimvar} threshold")
-                ds[var][ds[trimvar].isnull()] = np.nan
-
-                if var != trimvar:
-                    notetxt = f"Values filled using valid {trimvar} threshold. "
-
-                    ds = utils.insert_note(ds, var, notetxt)
-
-    return ds
+# remove qaqc defs from exo.py
```
Comment (dnowacki-usgs) [inline stglib/exo.py:603]: This comment is helpful for review but should probably be removed

File: stglib/eco.py [hunk 4]
```diff
@@ -269,38 +269,23 @@ def eco_qaqc(ds):
...
-
-        notetxt = (
-            "Values filled where standard deviation greater than %f "
-            "units. " % ds.attrs[var + "_std_max"]
-        )
-
-        ds = utils.insert_note(ds, var, notetxt)
-
-    return ds
+# relocate trim_max_std() to qaqc.py
```
Comment (dnowacki-usgs) [inline stglib/eco.py:291]: This comment is helpful for review but should probably be removed

**Review by dnowacki-usgs — CHANGES_REQUESTED** [review]

File: stglib/exo.py (hunk 2 above)
Comment (dnowacki-usgs) [reply stglib/exo.py:597]: Whoops, disregard. This was a formatting issue.

File: stglib/aqd/cdf2nc.py (hunk 1 above)
Comment (ssuttles-usgs) [reply stglib/aqd/cdf2nc.py:84]: accidently deleted. fixed in commit 2.

File: stglib/exo.py (hunk 2 above)
Comment (ssuttles-usgs) [reply stglib/exo.py:597]: Ok.

File: stglib/exo.py (hunk 3 above)
Comment (ssuttles-usgs) [reply stglib/exo.py]: removed.

File: stglib/eco.py (hunk 4 above)
Comment (ssuttles-usgs) [reply stglib/eco.py]: removed.

**Pull request merged**